frappe/automation/doctype/auto_repeat/auto_repeat.py

```diff
@@ -139,16 +138,18 @@ def validate_dates(self):
 			return
 
 		if self.end_date:
+			end_date = getdate(self.end_date)
+
 			self.validate_from_to_dates("start_date", "end_date")
-		if self.end_date == today():
-			frappe.throw(_("End Date cannot be today."))
 
-		if self.end_date == self.start_date:
-			frappe.throw(
-				_("{0} should not be same as {1}").format(
-					frappe.bold(_("End Date")), frappe.bold(_("Start Date"))
+			if end_date == getdate():
+				frappe.throw(_("End Date cannot be today."))
+			elif end_date == getdate(self.start_date):
```

Comment:
```suggestion
			
			if end_date == getdate(self.start_date):
```

Can leave it just `if` like earlier, since there's a throw in the previous line.